### PR TITLE
opt: Add "retry" option to zone logic test to prevent flakes

### LINKED ISSUE #35546: sqlccl: flake in logictestccl/testdata/logic_test/zone
Observed e.g. in https://teamcity.cockroachdb.com/viewLog.html?buildId=1168018&

```
------- Stdout: -------
=== PAUSE TestCCLLogic/5node-dist-opt/zone
=== CONT  TestCCLLogic/5node-dist-opt/zone
--- done: testdata/logic_test/zone with config 5node-dist-opt: 12 tests, 1 failures
--- total: 286 tests, 1 failures
test logs left over in: /tmp/logTestCCLLogic019160877
            logic.go:2313: 
                 
                testdata/logic_test/zone:70: EXECUTE p
                expected:
                    scan  ·      ·
                    ·     table  t@secondary
                    ·     spans  /10-/11
                but found (query options: "") :
                    scan  ·      ·
                    ·     table  t@primary
                    ·     spans  /10-/10/#
```

the spans are nearly equivalent.

if my eyes serve me well, this looks like the problem jordan solved earlier with #33350. Is it possible that a code path was not covered?

### PR DESCRIPTION
**Pull request by andy-kimball**
It can take a bit of time for gossip to update the zone after it's changed.
Use the "retry" option in the logic test to retry until gossip does the
update.

Fixes #35546

Release note: None

### CODE REVIEW

**Review by knz — APPROVED** [review]
thanks!

**Review by RaduBerinde — APPROVED** [review]

**Comment by knz** [comment]
Merging this because I'm running into tons of flakes.

bors r=knz,RaduBerinde

**Comment by andy-kimball** [comment]
Can't seem to get this to merge; not sure what's going on.

**Comment by knz** [comment]
it's blocked on separate issue/flake #35567.

will still retry

bors r=knz,RaduBerinde

**Comment by knz** [comment]
it looks like we have to investigate that test failure. maybe `master` is broken. Will look

**Comment by knz** [comment]
I analyzed the build log. The error around TestLogic is a red herring, the problem is that the TestLogic code outputs on buffered stdout and this interleaves with the stderr messages towards teamcity.  Will send disambiguating patch before we can investigate this.

**Comment by knz** [comment]
requeue after #35583 merges

**Comment by knz** [comment]
timeout has been bumped, let's try this again

bors r=knz,RaduBerinde

**Pull request merged**
